pkg/agent/multicluster/mc_route_controller.go

```diff
@@ -0,0 +1,465 @@
+	// Install Gateway flows only when the Node is the active Gateway.
+	if err := c.ofClient.InstallMulticlusterClassifierFlows(gw.Name, config.DefaultTunOFPort, true); err != nil {
+		return err
+	}
+	return c.addMCFlowsForCIImps(gw, nil)
+}
+
+func (c *MCRouteController) installFlowsOnRegularNode(gw *mcv1alpha1.Gateway) error {
+	if err := c.ofClient.InstallMulticlusterClassifierFlows(gw.Name, config.DefaultTunOFPort, false); err != nil {
+		return err
+	}
+	return c.addMCFlowsForCIImps(gw, nil)
+}
+
+func (c *MCRouteController) addMCFlowsForCIImps(activeGW *mcv1alpha1.Gateway, ciImp *mcv1alpha1.ClusterInfoImport) error {
+	var err error
+	var allciImports []*mcv1alpha1.ClusterInfoImport
+	if ciImp != nil {
+		allciImports = append(allciImports, ciImp)
+	} else {
+		allciImports, err = c.ciImportLister.ClusterInfoImports(c.namespace).List(labels.Everything())
+		if err != nil {
+			return err
+		}
+		if len(allciImports) == 0 {
+			klog.V(2).InfoS("No remote ClusterInfo imported, do nothing")
+			return nil
+		}
+	}
+
+	var tunnelPeerIP net.IP
+	for _, ciImport := range allciImports {
+		tunnelPeerIP = getPeerGatewayIP(ciImport.Spec)
+		if tunnelPeerIP == nil {
+			klog.V(2).InfoS("No valid peer Gateway IP from ClusterInfoImport, skip updating openflow rules",
+				"clusterinfoimport", klog.KObj(ciImport), "node", c.nodeConfig.Name)
+			continue
+		}
+
+		klog.InfoS("Adding/Updating remote Gateway Node flows for Multi-cluster", "gateway", klog.KObj(activeGW),
+			"node", c.nodeConfig.Name, "peer", tunnelPeerIP)
+		allCIDRs := []string{ciImport.Spec.ServiceCIDR}
+		peerConfigs, err := generatePeerConfigs(allCIDRs, tunnelPeerIP)
+		if err != nil {
+			klog.ErrorS(err, "Parse error for serviceCIDR from remote cluster", "clusterinfoimport", ciImport.Name, "gateway", activeGW.Name)
+			return err
+		}
+		if activeGW.Name == c.nodeConfig.Name {
+			klog.V(2).InfoS("Adding/Updating flows to remote Gateway Node for Multi-cluster traffic", "clusterinfoimport", ciImport.Name, "cidrs", allCIDRs)
+			localGatewayIP := net.ParseIP(activeGW.GatewayIP)
+			if err := c.ofClient.InstallMulticlusterGatewayFlows(
+				activeGW.Name+ciImport.Spec.ClusterID,
+				peerConfigs,
+				tunnelPeerIP,
+				localGatewayIP); err != nil {
+				return fmt.Errorf("failed to install flows to remote Gateway in ClusterInfoImport %s: %v", ciImport.Name, err)
+			}
+		} else {
+			klog.V(2).InfoS("Adding/Updating flows to the local active Gateway for Multi-cluster traffic", "clusterinfoimport", ciImport.Name, "cidrs", allCIDRs)
+			tunnelPeerIP := net.ParseIP(activeGW.InternalIP)
+			if err := c.ofClient.InstallMulticlusterNodeFlows(
+				activeGW.Name+ciImport.Spec.ClusterID,
+				peerConfigs,
+				tunnelPeerIP); err != nil {
+				return fmt.Errorf("failed to install flows to Gateway %s: %v", activeGW.Name, err)
+			}
+		}
+
+		c.installedCIImports.Add(ciImport)
+	}
+	c.installedActiveGWName = activeGW.Name
```

Comment:
Why we change installedActiveGWName here?

Reply:
It's stale code, removed